content/blog/chess-knight-problem/index.md

```diff
@@ -0,0 +1,129 @@
+---
+{
+  title: "Chess Knight Problem: a quick and dirty solution in JavaScript",
+  description: "Here I present a quick and dirty solution to a common interview question where the solution is not nearly as complex as it may first appear.",
+  published: "2020-04-29T12:27:06.284Z",
+  authors: ["thodges314"],
+  tags: ["javascript", "computer science", "interview"],
+  attached: [],
+  license: "cc-by-4",
+}
+---
+
+This is a classic problem in Computer Science that is said to come up in various forms on programming interviews - given an 8 x 8 chessboard, a selected starting square, and a selected ending square, what is the minimum number of legal moves are required to get a knight piece from the starting square to the ending square? Note that this is _not_ the same as Knight's Tour, a much more involved problem where we try to find a path around the board such that the knight touches every square once without repeat.
+
+For some background - a chess knight moves in an L-shaped pattern - two up and one to the right, two to the left and one up, and so on. The knight jumps from it's starting location to it's ending location.
```

Comment:
```suggestion
In case you're unaware of the rules of Chess, a chess knight is only allowed to move in an L-shaped pattern (two up and one to the right, two to the left and one up, that sort of thing). The knight jumps from it's starting location to it's ending location.
```

Reply:
I'd rather leave this one.